src/gzip_wrapper.h

```diff
@@ -51,8 +66,33 @@ inline std::string Gzip(const std::string& data_to_compress) {
       return compressed;
     }
   }
-  ALOG("ERROR", res, "while gzipping with zlib.", z.msg ? z.msg : "");
-  return std::string();
+  throw GzipErrorException(res, z.msg);
+}
+
+// Returns empty string on error.
+inline std::string Gunzip(const std::string& data_to_decompress)
+{
+  z_stream z = {};
+  int res = ::inflateInit2(&z, 16 + MAX_WBITS);
+  if (Z_OK == res) {
+    z.next_in = const_cast<Bytef*>(reinterpret_cast<const Bytef*>(data_to_decompress.data()));
+    z.avail_in = data_to_decompress.size();
+    std::string decompressed;
+    char buffer[std::min(data_to_decompress.size(), kGzipBufferSize)];
+    do {
+      z.next_out = reinterpret_cast<Bytef*>(buffer);
+      z.avail_out = sizeof(buffer) / sizeof(buffer[0]);
```

Comment:
AAAAA! Using a non-standard extension for dynamically-sized arrays and then assuming `sizeof` does it?

`std::vector<char>` would be so much safer here.